addons/persistence/mongodb-persistence-addon/src/main/java/org/kie/kogito/mongodb/utils/CommonUtils.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.kogito.mongodb.utils;
+
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+import org.bson.Document;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.codecs.pojo.PojoCodecProvider;
+import org.jbpm.marshalling.impl.ProcessInstanceDocument;
+import org.kie.kogito.mongodb.model.ProcessInstanceModel;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+
+public class CommonUtils {
+
+	private static MongoDatabase mongoDatabase;
+	private static MongoCollection<ProcessInstanceModel> collection;
+	private static final String KOGITO_STORE = "Kogito_store";
+
+	private static ObjectMapper MAPPER = new ObjectMapper();
+
+	public static MongoCollection<ProcessInstanceModel> getCollection(MongoClient mongoClient, String processId) {
+
+		PojoCodecProvider provider = PojoCodecProvider.builder().register(ProcessInstanceModel.class)
+				.register(org.jbpm.marshalling.impl.ProcessInstanceDocument.class).automatic(true)
+				.build();
+		CodecRegistry registry = fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+				fromProviders(provider));
+		mongoDatabase = mongoClient.getDatabase(KOGITO_STORE).withCodecRegistry(registry);
+		collection = mongoDatabase.getCollection(processId + "_store", ProcessInstanceModel.class)
```

Comment:
I followed the infinispan addon where a separate cache for each process is created. I can certainly change it to a single collection if needed.

Reply:
As MongoDB collections are analogous to traditional SQL DB tables, the decision is similar to whether using single or multiple collections/tables to keep process instances (i.e. rows). From Kogito perspective, it dose "not" reply on a consistent process data model, for example, it can include arbitrary business domain models. Therefore, a strategy of using dedicate collection per process would provide flexibility to apply indices targeted to attributes from various domain model. 
@xiezhang7 Do you have different thoughts on this?

Reply:
In this case, given the amount of records in each collection, it doesn't make much sense to use index.

Reply:
The collection will be created per process definition  (e.g it will create collection UserTask for below process and corresponding process instances will be stored into it and will be retrieved using process instance id.)
 https://github.com/RHsyseng/kogito-runtimes/blob/e6daad2a0c1a4c24f1af22ebcf23c8d0969f1070/addons/persistence/mongodb-persistence-addon/src/test/resources/BPMN2-UserTask.bpmn2#L15

It will also help in purging the collection if the process is deleted.

Reply:
Got it. Thanks.